### ISSUE TITLE
Feature Request: Limit Links

### ISSUE THREAD

**Issue by Cerberooo** [stevenvachon/broken-link-checker#97]
What about some new rules: 
* Add a total limit of checked links (eq. 10, 100 ...) 
* Add a optional limit for the recursive depth (level 1, 2 ...) 

When on of the limits is reached the blc terminates.

I would like to use the broken-link-checker for a tool and want to avoid, that the blc is running too long.

**Comment by neorganic** [comment]
I fully agree - these two features would be a great addition for the API. Of course at the same time this is something you can handle on your own part, but having this out of the box would be really nice.

**Comment by stevenvachon** [comment]
Would an overall timeout be better than a limited link count?

**Comment by Cerberooo** [comment]
Both sides have their advantages.
